service/history/workflowTaskHandler.go

```diff
@@ -236,6 +242,54 @@ func (handler *workflowTaskHandlerImpl) handleCommand(ctx context.Context, comma
 	}
 }
 
+func (handler *workflowTaskHandlerImpl) handleMessages(
+	ctx context.Context,
+	messages []*protocolpb.Message,
+) error {
+	if err := handler.attrValidator.validateMessages(
+		messages,
+	); err != nil {
+		return err
+	}
+
+	for _, message := range messages {
+		err := handler.handleMessage(ctx, message)
+		if err != nil || handler.stopProcessing {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (handler *workflowTaskHandlerImpl) handleMessage(ctx context.Context, message *protocolpb.Message) error {
+
+	if types.Is(message.GetBody(), (*updatepb.Acceptance)(nil)) {
+		var acceptance updatepb.Acceptance
+		err := types.UnmarshalAny(message.GetBody(), &acceptance)
+		if err != nil {
+			return err
+		}
+		return handler.handleMessageAcceptWorkflowExecutionUpdate(ctx, message.GetProtocolInstanceId(), &acceptance)
+	} else if types.Is(message.GetBody(), (*updatepb.Response)(nil)) {
+		var response updatepb.Response
+		err := types.UnmarshalAny(message.GetBody(), &response)
+		if err != nil {
+			return err
+		}
+		return handler.handleMessageCompleteWorkflowExecutionUpdate(ctx, message.GetProtocolInstanceId(), &response)
+	} else if types.Is(message.GetBody(), (*updatepb.Rejection)(nil)) {
+		var rejection updatepb.Rejection
+		err := types.UnmarshalAny(message.GetBody(), &rejection)
+		if err != nil {
+			return err
+		}
+		return handler.handleMessageRejectWorkflowExecutionUpdate(ctx, message.GetProtocolInstanceId(), &rejection)
+	}
+
+	return nil
+}
```

Comment:
TODO but later - we definitely want to do this dispatch within a protocol state machine. On the sdk-side we find the protocol object by protocol_instance_id and then just call HandleMessage.